sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/LruCacheTests.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro.Tests
+{
+    public class LruCacheTests
+    {
+        [Test]
+        public void CacheCapacityRespected()
+        {
+            var cache = new LruCache<string, int>(10);
+            for (int i = 0; i < 20; i++)
+            {
+                cache.AddOrUpdate(i.ToString(), i);
+            }
+
+            for (int i = 0; i < 10; i++)
+            {
+                Assert.IsFalse(cache.TryGet(i.ToString(), out _));
+            }
+
+            for (int i = 11; i < 20; i++)
+            {
+                Assert.IsTrue(cache.TryGet(i.ToString(), out var value));
+                Assert.AreEqual(i, value);
+            }
+        }
+
+        [Test]
+        public void CacheOrderingRespected()
```

Comment:
nite: since the cache doesn't exist the least recently used item, the ordering is opaque.
```suggestion
        public void CacheEvictsLeastRecentlyAccessedItemFirst()
```